src/com/cafeto/training/LazySingletonTestAccess.java

```diff
@@ -0,0 +1,45 @@
+package com.singleton;
+
+import com.acces.IDataAccess;
+
+
+import org.mockito.Mockito;
+
+/**
+ * This test support class that set and clear a mock instance.
+ */
+public class LazySingletonTestAccess
+{
+
+    /**
+     * This method instantiate a lazySingleton class and set the IDataAccess object received as parameter
+     *
+     * @param dataAccess
+     * @return IDataAccess object.
+     */
+    public IDataAccess setMockInstance(IDataAccess dataAccess)
+    {
+        lazySingleton instance= Mockito.mock(lazySingleton.class).getLazySingleton();
```

Comment:
you can't mock the LazySingleton class, we are trying to test it, here you should do the LazySingleton.getInstance().dataAccess = mockDataAccess;